src/compiler/checker.ts

```diff
@@ -2773,11 +2770,23 @@ namespace ts {
         }
 
         function getTargetOfImportSpecifier(node: ImportSpecifier | BindingElement, dontResolveAlias: boolean): Symbol | undefined {
-            const resolved = getExternalModuleMember(isBindingElement(node) ? getRootDeclaration(node) as VariableDeclaration : node.parent.parent.parent, node, dontResolveAlias);
+            const root = isBindingElement(node) ? getRootDeclaration(node) as VariableDeclaration : node.parent.parent.parent;
+            const commonJSPropertyAccess = getCommonJSPropertyAccess(root);
+            const resolved = getExternalModuleMember(root, commonJSPropertyAccess || node, dontResolveAlias);
+            const name = node.propertyName || node.name;
+            if (commonJSPropertyAccess && resolved && isIdentifier(name)) {
+                return getPropertyOfType(getTypeOfSymbol(resolved), name.escapedText);
+            }
             markSymbolOfAliasDeclarationIfTypeOnly(node, /*immediateTarget*/ undefined, resolved, /*overwriteEmpty*/ false);
             return resolved;
         }
 
+        function getCommonJSPropertyAccess(node: Node) {
+            if(isVariableDeclaration(node) && node.initializer && isPropertyAccessExpression(node.initializer)) {

```

Comment:
```suggestion
            if (isVariableDeclaration(node) && node.initializer && isPropertyAccessExpression(node.initializer)) {

```

Reply:
I tried to commit your suggestion directly but github kept claiming the PR was recently updated =(